upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -441,29 +441,45 @@ func (c *ApplyClusterCmd) Run() error {
 	l.WorkDir = c.OutDir
 	l.ModelStore = modelStore
 
-	assetBuilder := assets.NewAssetBuilder()
+
+
+	assetBuilder := assets.NewAssetBuilder(cluster.Spec.Assets)
 
 	iamLifecycle := lifecyclePointer(fi.LifecycleSync)
 	networkLifecycle := lifecyclePointer(fi.LifecycleSync)
 	clusterLifecycle := lifecyclePointer(fi.LifecycleSync)
+	// TODO - for @justinsb to look at
+	//assetTransferLifecycle = lifecyclePointer(fi.LifecycleSync)
 
 	switch c.Phase {
 	case Phase(""):
 	// Everything ... the default
 
+		/*
+		// TODO
+	case PhaseAssetsUpdate:
+	 	assetTransferLifecycle = lifecyclePointer(fi.LifecycleSync)
+		iamLifecycle = lifecyclePointer(fi.LifecycleIgnore)
+		networkLifecycle = lifecyclePointer(fi.LifecycleIgnore)
+		clusterLifecycle = lifecyclePointer(fi.LifecycleIgnore)
+		*/
 	case PhaseIAM:
+		//assetTransferLifecycle = lifecyclePointer(fi.LifecycleIgnore)
 		networkLifecycle = lifecyclePointer(fi.LifecycleIgnore)
 		clusterLifecycle = lifecyclePointer(fi.LifecycleIgnore)
 
 	case PhaseNetwork:
+		//assetTransferLifecycle = lifecyclePointer(fi.LifecycleIgnore)
 		iamLifecycle = lifecyclePointer(fi.LifecycleIgnore)
 		clusterLifecycle = lifecyclePointer(fi.LifecycleIgnore)
 
 	case PhaseCluster:
 		if c.TargetName == TargetDryRun {
+			//assetTransferLifecycle = lifecyclePointer(fi.LifecycleExistsAndWarnIfChanges)
 			iamLifecycle = lifecyclePointer(fi.LifecycleExistsAndWarnIfChanges)
 			networkLifecycle = lifecyclePointer(fi.LifecycleExistsAndWarnIfChanges)
 		} else {
+			//assetTransferLifecycle = lifecyclePointer(fi.LifecycleIgnore)
```

Comment:
`LifecycleExistsAndValidates` I believe - we want to know if images aren't in place.